### PR TITLE
Now pulling indexStats information

### PR DESCRIPTION
**Pull request by ciudilo**
Added output from $indexStats.
Using runCommand instead of aggregate in order to catch errors when server does not support $indexStats or we don't have correct permissions.

### CODE REVIEW

**Comment by Dabz** [comment]
Looks good to me for merging (but I don't have write access for the merge :) )

File: getMongoData/getMongoData.js [hunk 1]
```diff
@@ -302,6 +302,27 @@ function printDataInfo(isMongoS) {
...
+                                  {$group: {_id: "$key", stats: {$push: {accesses: "$accesses.ops", host: "$host", since: "$accesses.since"}}}},
+                                  {$project: {key: "$_id", stats: 1, _id: 0}}
+                                ]});
+
+                              if (res.hasOwnProperty('result')) {
+                                res.result.forEach(
+                                  function(d){
+                                    d.stats.map(
+                                      function(d){
+                                        return d.since = d.since.toUTCString();
```
Comment (devkev) [inline getMongoData/getMongoData.js]: The `return` here isn't great, even though in this case it doesn't hurt.  Please remove it and change the above `d.stats.map` to `d.stats.forEach`.

File: getMongoData/getMongoData.js [hunk 2]
```diff
@@ -302,6 +302,27 @@ function printDataInfo(isMongoS) {
                     }
                     printInfo('Indexes',
                               function(){return db.getSiblingDB(mydb.name).getCollection(col).getIndexes()}, section);
+                    printInfo('Index Stats',
+                              function(){var res = db.getSiblingDB(mydb.name).runCommand( {
```
Comment (devkev) [inline getMongoData/getMongoData.js]: Please fix the indenting here - start a new line for the function contents, and ensure that all lines in this function are aligned.

File: getMongoData/getMongoData.js [hunk 3]
```diff
@@ -302,6 +302,27 @@ function printDataInfo(isMongoS) {
...
+                                aggregate: col,
+                                pipeline: [
+                                  {$indexStats: {}},
+                                  {$group: {_id: "$key", stats: {$push: {accesses: "$accesses.ops", host: "$host", since: "$accesses.since"}}}},
```
Comment (devkev) [inline getMongoData/getMongoData.js]: Can you explain the need for this?  Presumably it's for when running against a sharded cluster via mongos, and obtaining results from several shards?

Comment (ciudilo) [inline getMongoData/getMongoData.js]: Correct, the grouping is there in order to make output more readable when running against a sharded cluster.

File: getMongoData/getMongoData.js (hunk 1 above)
Comment (ciudilo) [inline getMongoData/getMongoData.js]: done

File: getMongoData/getMongoData.js (hunk 2 above)
Comment (ciudilo) [inline getMongoData/getMongoData.js]: done

**Comment by devkev** [comment]
LGTM, thanks! :+1:

**Pull request merged**